src/Hooks/useDeeplinkUrl.ts

```diff
@@ -0,0 +1,28 @@
+import {useEffect, useState} from 'react';
+import {Linking} from 'react-native';
+
+export const useDeeplinkUrl = () => {
+  const [deeplinkUrl, setDeeplinkUrl] = useState<string | null>(null);
+
+  const EVENT_NAME = 'url';
+
+  useEffect(() => {
+    // If the app is already open, the app is foregrounded and a Linking 'url' event is fired
+    Linking.addEventListener(EVENT_NAME, ({url}) => {
+      setDeeplinkUrl(url);
+    });
+
+    // If the app is not already open, it is opened and the url is passed in as the initialURL
+    const getUrlAsync = async () => {
+      const initialUrl = await Linking.getInitialURL();
```

Comment:
This function returning same value all the time. Did you test it 
1) opened with deep link (closed app)
2) send a new deeplink event
conditions? 
first step trigger getInitialURL but second step trigger eventListener. I am not sure about URL's final value